pkg/controllers/pod/pod_controller.go

```diff
@@ -214,14 +214,15 @@ func (r *PodReconciler) CollectGarbage(ctx context.Context) {
 
 	PodSet := sets.New[string]()
 	for _, pod := range podList.Items {
-		PodSet.Insert(string(pod.UID))
+		subnetPortID := r.SubnetPortService.BuildSubnetPortId(&pod.ObjectMeta)
+		PodSet.Insert(subnetPortID)
 	}
 
 	diffSet := nsxSubnetPortSet.Difference(PodSet)
 	for elem := range diffSet {
 		log.V(1).Info("GC collected Pod", "UID", elem)
```

Comment:
Could you please also change the log? `UID` - > `NSXSubnetPortID`

Reply:
updated.